### ISSUE TITLE
vscode-jest activates for all folders in a multi-root workspace if any folder contains jest-related files

### ISSUE THREAD

**Issue by nicksnyder** [jest-community/vscode-jest#582]
vscode-jest activates for all roots in a multi-root workspace if [any folder contains jest-related files](https://sourcegraph.com/github.com/jest-community/vscode-jest/-/blob/package.json#L25-29). When it activates on folders that don't have jest files, it spams a bunch of error messages.

### Expected Behavior

Don't show errors on roots where there are no jest files.

### Actual Behavior

Shows errors on roots that don't use jest.

![Screen Shot 2020-05-05 at 1 19 15 PM (1)](https://user-images.githubusercontent.com/754768/81352298-08b9e880-907b-11ea-82d7-7ef988db8c3a.png)

### Workaround

It seems like only way to reduce scope would be to exhaustively list all folders where Jest should be disabled with `jest.disabledWorkspaceFolders` - there’s no way to specify a whitelist of folders where Jest should run (which would be better).

**Comment by connectdotz** [comment]
While this feature is definitely nice to have, it is the expected behavior for the current implementation: instead of doing a deep search for jest files on each folder, it took a simpler approach by providing a customize-option... As the error panel indicated above, you can skip these folders by adding `"jest.disabledWorkspaceFolders": ['interviews', 'deploy-sourcegraph']` to .vscode/settings.json or the equivalent.

PR is welcome.

**Comment by peterldowns** [comment]
@connectdotz would it be possible to read the `disabledWorkspaceFolders` from the `workspace` settings, and not just the user settings?

Right now if I have a `myproject.code-workspace` like this:
```json
{
  "folders": [
    { "path": "backend" },
    { "path": "client/web" }
  ],
  "settings": {
    "jest.runAllTestsFirst": true,
    "jest.debugMode": true
  },
  "jest.debugCodeLens.showWhenTestStateIn": [
      "fail",
      "unknown",
      "pass",
      "skip"
  ],
  "jest.disabledWorkspaceFolders": [
      "backend"
  ],
}
```
and launch it like this:
```bash
code myproject.code-workspace
```
then Jest attempts to start for the `backend` folder:
![image](https://user-images.githubusercontent.com/824173/81624491-2785eb00-93ab-11ea-94a3-50a47e0909c9.png)

but if I update my user `settings.json`, to contain

```json
"jest.disabledWorkspaceFolders": [
  "backend"
]
```
then Jest (correctly) does not attempt to start for the `backend` folder.

I'm not sure why I'm seeing this behavior, since [the code to read the project config](https://github.com/jest-community/vscode-jest/blob/745e6035c7138fe358a03d01be97a520869de1bf/src/extensionManager.ts#L198) looks right, but I'm not very familiar with the VisualStudio Code APIs.

**Comment by decademoon** [comment]
Why do we have to opt out workspace folders instead of opt in? I don't understand your justification @connectdotz.

**Comment by connectdotz** [comment]
> Why do we have to opt out workspace folders instead of opt in? I don't understand your justification @connectdotz

hmmm... I don't think I am advocating opt-out is better than opt-in, just stating the current implementation and how to work around it. I am agreeing with this issue and therefore marked it as a "feature"/"enhancement". But judging from the thumb-down for my previous comment, you are not the only person getting the wrong impression 😓 ... My bad, but the invitation for PR stands, we are welcoming enhancement to make this experience better.

**Comment by jonnytest1** [comment]
https://github.com/jest-community/vscode-jest/pull/896 how about this ? :)

**Comment by connectdotz** [comment]
close this issue via #896, which should be available in the next pre-release (v4.7). Thanks @jonnytest1 !
